packages/components/src/popover/index.tsx

```diff
@@ -569,17 +569,16 @@ const UnforwardedPopover = (
 
 	if ( shouldRenderWithinSlot ) {
 		content = <Fill name={ slotName }>{ content }</Fill>;
+	} else if ( ! inline ) {
+		content = createPortal( content, getPopoverFallbackContainer() );
 	}
 
-	if ( ! hasAnchor ) {
-		content = <span ref={ anchorRefFallback }>{ content }</span>;
-	}
-
-	if ( shouldRenderWithinSlot || inline ) {
-		return content;
-	}
-
-	return createPortal( content, getPopoverFallbackContainer() );
+	return (
+		<>
+			{ ! hasAnchor && <span ref={ anchorRefFallback } /> }
+			{ content }
+		</>
+	);
```

Comment:
Style nit: most of the time the anchor is specified, so let's avoid rendering the redundant fragment by writing it as:
```js
if ( hasAnchor ) {
  return content;
}
return <><span/>{ content }</>;
```

Reply:
Just to understand as I catch up with the changes happened to Popover:

previously, `{ content }` was rendered as a child of the anchor `span`, but it's not rendered as a sibling. Is that on purpose for this PR ?

Reply:
Previously, the `{ content }`, i.e., the popover element (used as `floating` by Floating UI), would be rendered as a child of the `span`, and it would be absolutely positioned, so it's not part of the layout flow. So, layout-wise, the `<span>` is empty, and Floating UI ensures that the popover is positioned correctly relative to the empty `<span>` anchor.

Now, the `{ content }` is rendered in a portal element, at a completely different location. The `<span>` is empty not only layout-wise, but also DOM-markup-wise. Again, Floating UI is managing the position.

It's true that `<span>{ content }</span>` would also work, because `content` is a portal, it doesn't affect the DOM structure of the `span` at all. The only difference is whether the `span` React element receives the events dispatched withing the portal. But as the `span` has no listeners, it doesn't matter.